### PR TITLE
fix: remove hover state on mobile

### PR DESCRIPTION
**Pull request by jianwei1**
# Description

### Issue

The hover style(for the stepblocknodemenu, and the red arrow) was breaking the mobile touch behaviour on cards, as it was messing with the touch events. On iphone, the first touch would make the menu and arrow appear, and you then needed a second touch to select the card. On android, it was the opposite. This ticket keeps the hover behaviour on PC, while on mobile it will both display the menu/arrow and select the card with a single touch.

[Link to Basecamp Todo](https://3.basecamp.com/3105655/buckets/37071143/todos/7322963113)

### Solution

Check if user has a touch device, if so use touch style, else use desktop styles (hover). Solution to detect touchscreen was [found here](https://stackoverflow.com/questions/56324813/how-to-detect-touch-device-in-2019). 

# Additional information

On stepblocknode.tsx, the original code wrapped the card and any action blocks(eg buttons on the card) inside one basenode. This meant that if you hovered over an action button, the red arrow would appear on the card. To fix this I moved the action buttons outside of that basenode.

This ticket also solves the [need to double touch to add card](https://3.basecamp.com/3105655/buckets/37071143/todos/7311724727) on IOS bug

### CODE REVIEW

File: apps/journeys-admin/src/components/Editor/Slider/JourneyFlow/nodes/BaseNode/BaseNode.tsx [hunk 1]
```diff
@@ -29,17 +30,30 @@ export function BaseNode({
...
+    },
+    ':hover .arrow': {
+      visibility: 'visible'
+    }
+  }
+
+  const touchStyle = {
+    '.arrow': {
+      visibility:
+        typeof selected === 'boolean' && selected ? 'visible' : 'hidden'
```
Comment (GeronimoJohn) [inline apps/journeys-admin/src/components/Editor/Slider/JourneyFlow/nodes/BaseNode/BaseNode.tsx:47]: is this for a type check? 'type of selected === 'boolean'. I think we can get away this via doing `selected === true ? 'visible' : 'hidden'`

File: apps/journeys-admin/src/components/Editor/Slider/JourneyFlow/nodes/BaseNode/BaseNode.tsx [hunk 2]
```diff
@@ -61,24 +75,59 @@ export function BaseNode({
...
+            onConnect={onSourceConnect}
+            {...sourceHandleProps}
+            sx={{
+              width: 7.5,
+              height: 7.5,
+              background: 'white',
+              border:
+                selected !== false
+                  ? '2px solid #c52d3aff'
+                  : '2px solid #aaacbb',
```
Comment (GeronimoJohn) [inline apps/journeys-admin/src/components/Editor/Slider/JourneyFlow/nodes/BaseNode/BaseNode.tsx:93]: If available and if in the right context. Meaning the right ThemeProvider. Can we refactor this to be using the color token

File: apps/journeys-admin/src/components/Editor/Slider/JourneyFlow/nodes/BaseNode/BaseNode.tsx [hunk 3]
```diff
@@ -61,24 +75,59 @@ export function BaseNode({
...
+            id={id}
+            type="source"
+            data-testid="BaseNodeBottomHandle"
+            position={Position.Bottom}
+            onConnect={onSourceConnect}
+            {...sourceHandleProps}
+            sx={{
+              width: 7.5,
+              height: 7.5,
+              background: 'white',
```
Comment (GeronimoJohn) [inline apps/journeys-admin/src/components/Editor/Slider/JourneyFlow/nodes/BaseNode/BaseNode.tsx:89]: refactor to color token please

File: apps/journeys-admin/src/components/Editor/Slider/JourneyFlow/nodes/BaseNode/BaseNode.tsx [hunk 4]
```diff
@@ -61,24 +75,59 @@ export function BaseNode({
...
+              border:
+                selected !== false
+                  ? '2px solid #c52d3aff'
+                  : '2px solid #aaacbb',
+              outline: '1px solid',
+              outlineColor: 'white',
```
Comment (GeronimoJohn) [inline apps/journeys-admin/src/components/Editor/Slider/JourneyFlow/nodes/BaseNode/BaseNode.tsx:95]: color token if available

File: apps/journeys-admin/src/components/Editor/Slider/JourneyFlow/nodes/BaseNode/BaseNode.tsx [hunk 5]
```diff
@@ -61,24 +75,59 @@ export function BaseNode({
...
+                data-testid="BaseNodeDownwardArrowIcon"
+                className="arrow"
+                style={{
+                  display: 'flex',
+                  position: 'absolute',
+                  borderRadius: '50%',
+                  color: 'white',
+                  fontSize: 'large',
+                  top: '50%',
+                  backgroundColor: '#c52d3aff',
```
Comment (GeronimoJohn) [inline apps/journeys-admin/src/components/Editor/Slider/JourneyFlow/nodes/BaseNode/BaseNode.tsx:122]: color token

File: apps/journeys-admin/src/components/Editor/Slider/JourneyFlow/nodes/BaseNode/BaseNode.tsx [hunk 6]
```diff
@@ -61,24 +75,59 @@ export function BaseNode({
...
+          >
+            {isSourceConnectable === 'arrow' && (
+              <ArrowDownwardRoundedIcon
+                data-testid="BaseNodeDownwardArrowIcon"
+                className="arrow"
+                style={{
+                  display: 'flex',
+                  position: 'absolute',
+                  borderRadius: '50%',
+                  color: 'white',
```
Comment (GeronimoJohn) [inline apps/journeys-admin/src/components/Editor/Slider/JourneyFlow/nodes/BaseNode/BaseNode.tsx:119]: color token

File: apps/journeys-admin/src/components/Editor/Slider/JourneyFlow/nodes/BaseNode/BaseNode.tsx [hunk 7]
```diff
@@ -61,24 +75,59 @@ export function BaseNode({
...
+                  fontSize: 'large',
+                  top: '50%',
+                  backgroundColor: '#c52d3aff',
+                  left: '50%',
+                  transform: 'translate(-50%, -50%)',
+                  pointerEvents: 'none'
+                }}
+              />
+            )}{' '}
```
Comment (GeronimoJohn) [inline apps/journeys-admin/src/components/Editor/Slider/JourneyFlow/nodes/BaseNode/BaseNode.tsx:128]: can remove the extra space {'  '}

File: apps/journeys-admin/src/components/Editor/Slider/JourneyFlow/nodes/StepBlockNode/StepBlockNode.tsx [hunk 8]
```diff
@@ -57,205 +57,244 @@ export function StepBlockNode({ id }: NodeProps): ReactElement {
...
-          >
-            <StepBlockNodeMenu className="fab" step={step} />
-            <Card
+      <BaseNode
+        id={step.id}
+        isTargetConnectable
+        isSourceConnectable="arrow"
+        selected={isSelected}
+        onSourceConnect={handleSourceConnect}
+        sourceHandleProps={{
```
Comment (GeronimoJohn) [inline apps/journeys-admin/src/components/Editor/Slider/JourneyFlow/nodes/StepBlockNode/StepBlockNode.tsx:110]: remove?

File: apps/journeys-admin/src/components/Editor/Slider/JourneyFlow/nodes/StepBlockNode/StepBlockNode.tsx [hunk 9]
```diff
@@ -57,205 +57,244 @@ export function StepBlockNode({ id }: NodeProps): ReactElement {
...
+          sx: {
+            //  bottom: actionBlocks.length > 0 ? 35 : 0
+          }
+        }}
+      >
+        {({ selected }) => (
+          <Stack data-testid="spacingStack" alignItems="center">
+            <Box
+              onTouchStart={() => console.log('touch')}
```
Comment (GeronimoJohn) [inline apps/journeys-admin/src/components/Editor/Slider/JourneyFlow/nodes/StepBlockNode/StepBlockNode.tsx:119]: not needed

File: apps/journeys-admin/src/components/Editor/Slider/JourneyFlow/nodes/StepBlockNode/StepBlockNode.tsx [hunk 10]
```diff
@@ -57,205 +57,244 @@ export function StepBlockNode({ id }: NodeProps): ReactElement {
...
+        selected={isSelected}
+        onSourceConnect={handleSourceConnect}
+        sourceHandleProps={{
+          sx: {
+            //  bottom: actionBlocks.length > 0 ? 35 : 0
+          }
+        }}
+      >
+        {({ selected }) => (
+          <Stack data-testid="spacingStack" alignItems="center">
```
Comment (GeronimoJohn) [inline apps/journeys-admin/src/components/Editor/Slider/JourneyFlow/nodes/StepBlockNode/StepBlockNode.tsx:117]: remove data-testid if only was used for testing purposes. else if deemed needed follow the pattern for allocating data-testid

File: apps/journeys-admin/src/components/Editor/Slider/JourneyFlow/nodes/StepBlockNode/StepBlockNode.tsx [hunk 11]
```diff
@@ -57,205 +57,244 @@ export function StepBlockNode({ id }: NodeProps): ReactElement {
...
+                </CardContent>
+              </Card>
+            </Box>
           </Stack>
-        </Stack>
-      )}
-    </BaseNode>
+        )}
+      </BaseNode>
+
```
Comment (GeronimoJohn) [inline apps/journeys-admin/src/components/Editor/Slider/JourneyFlow/nodes/StepBlockNode/StepBlockNode.tsx:283]: can remove this space

**Review by GeronimoJohn — COMMENTED** [review]
There's just a few colors thats using hex. Are you able to just refactor those to be using the color token? Thought we might do this now. else it'll just be more tech debt for us

File: apps/journeys-admin/src/components/Editor/Slider/JourneyFlow/nodes/BaseNode/BaseNode.tsx (hunk 1 above)
Comment (jianwei1) [reply apps/journeys-admin/src/components/Editor/Slider/JourneyFlow/nodes/BaseNode/BaseNode.tsx:47]: selected?: 'descendant' | boolean

selected can also be a string, so I needed to type check that it was a boolean

**Review by GeronimoJohn — APPROVED** [review]

**Pull request merged**